core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2204,27 +2203,32 @@ class ReplicaManager(val config: KafkaConfig,
 
           completeDelayedFetchOrProduceRequests(tp)
 
-          // Create the local replica even if the leader is unavailable. This is required
```

Comment:
We seem to have dropped this logic. Do we not still need it?

Reply:
Yes. Good catch. To match the ZK implementation, we need to at least create the log when partition.makeFollower is not called.